FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -145,27 +151,57 @@ extension ComposeViewController {
     private func encryptAndSendMessage() -> Promise<Bool> {
         Promise<Bool> { [weak self] () -> Bool in
             guard let self = self else { return false }
-            guard let email = self.contextToSend.resipient, let text = self.contextToSend.message else {
+
+            guard self.contextToSend.recipients.isNotEmpty else {
+                assertionFailure("Recipients should not be empty. Fail in checking");
+                return false
+            }
+
+            guard let text = self.contextToSend.message else {
                 assertionFailure("Text and Email should not be nil at this point. Fail in checking");
                 return false
             }
-            let subject = self.input.subjectReplyTitle ?? self.contextToSend.subject ?? "(no subject)"
-            let lookupRes = try await(self.attesterApi.lookupEmail(email: email))
-            guard let recipientPubkey = lookupRes.armored else {
+
+            let subject = self.input.subjectReplyTitle
+                ?? self.contextToSend.subject
+                ?? "(no subject)"
+
+            let lookup = self.contextToSend.recipients.map {
+                self.attesterApi.lookupEmail(email: $0.email)
+            }
+
+            let lookupRes = try await(all(lookup))
+
+            let armoredKeys = lookupRes.map { $0.armored }
+
+            if armoredKeys.contains(nil) {
                 self.showAlert(message: "compose_no_pub_recipient".localized)
                 return false
             }
+
+            // TODO: ANTON -
             guard let myPubkey = self.dataManager.publicKey() else {
                 self.showAlert(message: "compose_no_pub_sender".localized)
                 return false
             }
-            let encrypted = self.encryptMsg(
-                pubkeys: [myPubkey, recipientPubkey],
-                subject: subject,
-                message: text,
-                email: email
-            )
-            try await(self.imap.sendMail(mime: encrypted.mimeEncoded))
+
+            let encryptedMessages = lookupRes.compactMap { (searchResult: PubkeySearchResult) -> CoreRes.ComposeEmail? in
+                guard let armored = searchResult.armored else { assertionFailure(); return nil }
+                return self.encryptMsg(
+                    pubkeys: [myPubkey, armored],
```

Comment:
This is wrong, there should be a single email message that is encrypted for all pubkeys, and mine